src/shared/hooks/__tests__/usePrefetch.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { queryByTestId } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+
+import './__mocks__/relListSupportMock';
+
+import usePrefetch from '../usePrefetch';
+
+const url = '/some/path';
+const url2 = '/some/other/path';
+
+describe('usePrefetch hook', () => {
+  test('no URL', async () => {
+    renderHook(() => usePrefetch());
+
+    const link = queryByTestId(document, 'prefetch');
+    expect(link).not.toBeNull();
+    expect(link.getAttribute('href')).toBeFalsy();
+  });
+
+  test('basic case', async () => {
+    renderHook(() => usePrefetch(url));
+
+    const link = queryByTestId(document, 'prefetch');
+    expect(link).not.toBeNull();
+    expect(link.getAttribute('rel')).toBe('prefetch');
+    expect(link.getAttribute('href')).toBe(url);
+  });
+
+  test('change of URL', async () => {
+    const { rerender } = renderHook((props) => usePrefetch(props.url), {
+      initialProps: { url },
+    });
+
+    const link = queryByTestId(document, 'prefetch');
+    expect(link).not.toBeNull();
+    expect(link.getAttribute('rel')).toBe('prefetch');
+    expect(link.getAttribute('href')).toBe(url);
+
+    rerender({ url: url2 });
```

Comment:
Just checking- there can be multiple prefetch `link`s at the same time right? not just one `link` with a changeable url?

Reply:
Yes, you can have multiple prefetch links.
Here in this test I can be sure there is only one because I'm using the hook once.
And I made sure to have the hook reuse the same link element it's using all along its lifecycle and to cleanup after itself on unmount for performance reasons